pyproject.toml

```diff
@@ -57,22 +66,26 @@ ml_runtime = [
     "catboost>=1.1.1",
     "requests-mock>=1.10.0",
     "tensorflow-hub>=0.12.0",
-    "transformers>=4.33, <4.34", # https://github.com/huggingface/transformers/issues/23352
-    "absl-py", # Needed for 3.11 transformers
-    "sentencepiece", # needed for some transformers stuff with tokenizer
+    "transformers>=4.34.1",
+    "sentencepiece",                                                                      # needed for some transformers stuff with tokenizer
     "torch>=2.0.0",
     "torchdata>=0.6.0",
-    "portalocker>=2.0.0", # Needed by torchdata for test_newspaper_classification_pytorch_dataset
+    "portalocker>=2.0.0",                                                                 # Needed by torchdata for test_newspaper_classification_pytorch_dataset
     "torchtext>=0.15.1",
-    "tensorflow-macos>=2.13.0, <2.14; sys_platform == 'darwin' and platform_machine == 'arm64'",
-    "tensorflow>=2.13, <2.14",
+    "tensorflow-macos==2.14.0; sys_platform == 'darwin' and platform_machine == 'arm64'",
+    "tensorflow==2.14.0",
     # tensorflow-text is only available on linux or mac, but not for arm64 nor windows
-    "tensorflow-text>=2.13, <2.14; python_version < '3.11' and (sys_platform == 'linux' or sys_platform == 'darwin') and platform_machine == 'x86_64'",
+    "tensorflow-text==2.14.0; python_version < '3.11' and (sys_platform == 'linux' or sys_platform == 'darwin') and platform_machine == 'x86_64'",
```

Comment:
Any way to avoid setting such specific versions? We would be imposing a hard constraint on users.

Reply:
Those deps are not installed when you install giskard, these are only for dev/testing purpose since they are under tool.pdm.dev-dependencies. Also, this is a way to get faster package resolution when updating the lock, especially since I discovered that pdm is not making a proper lock directly under macos, and that you have either to run it under docker or in linux non arm machine..

The actual prod requirements are under dependencies and server

Reply:
I see, I misread the diff